main/config.c

```diff
@@ -1221,6 +1266,14 @@ void ast_category_destroy(struct ast_category *cat)
 	ast_free(cat);
 }
 
+static void ast_config_included_files_destroy(struct ast_config_include *incl)
+{
+	struct ast_config_included_file *x;
+
+	while ((x = AST_LIST_REMOVE_HEAD(&incl->included_files, next)))
+		ast_free(x);
```

Comment:
As written, you would also need to free x->fn here, but if you follow my suggestion, you can leave this as is since there would only be one allocation.